src/main/java/uk/nhs/hee/tis/trainee/forms/service/impl/FormRPartAServiceImpl.java

```diff
@@ -110,4 +118,40 @@ public FormRPartADto getFormRPartAById(String id, String traineeTisId) {
     return mapper.toDto(formRPartA);
   }
 
+  /**
+   * Partial delete a form by id.
+   */
+  @Override
+  public FormRPartADto partialDeleteFormRPartAById(
+      String id, String traineeTisId, String[] fixedFields) {
+    log.info("Request to partial delete FormRPartA by id : {}", id);
+
+    try {
+      FormRPartA formRPartA = repository.findByIdAndTraineeTisId(UUID.fromString(id), traineeTisId)
+          .orElse(null);
+
+      if (formRPartA != null) {
+        ObjectMapper objectMapper = new ObjectMapper().findAndRegisterModules();
+        JsonNode jsonForm = objectMapper.valueToTree(formRPartA);
+
+        for (Iterator<String> fieldIterator = jsonForm.fieldNames(); fieldIterator.hasNext(); ) {
+          String fieldName = fieldIterator.next();
+
+          if (!Set.of(fixedFields).contains(fieldName)) {
```

Comment:
`Set.of(fixedFields)` here will create a new `Set` on every loop iteration, I recommend initializing the set outside of the loop. Alternatively, changing the method parameter to require a Set might be clearer.